### PR TITLE
Upgrade apollo client to fix app error

### PR DESCRIPTION
**Pull request by chiaberry**
## Associated issues

https://github.com/cityofaustin/atd-data-tech/issues/16931

The error message that was appearing was due to something in an apollo warning: https://github.com/apollographql/apollo-client/issues/11544

## Testing
**URL to test:** 

https://deploy-preview-1349--atd-moped-main.netlify.app/moped/projects/1862?tab=map

**Steps to test:**

Move one of the components to another project. 
Navigate to that project
use the back button to return to original project. 
No Crash!

---
#### Ship list
- [ ] Code reviewed 
- [ ] Product manager approved
- [ ] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

File: moped-editor/src/views/projects/projectView/ProjectSummary/TagsSection.js [hunk 1]
```diff
@@ -164,7 +164,7 @@ const TagsSection = ({ projectId }) => {
           >
             <Grid container spacing={1}>
               {data.moped_proj_tags.map((tag) => (
-                <Grid item>
+                <Grid item key={tag.id}>
```
Comment (chiaberry) [inline moped-editor/src/views/projects/projectView/ProjectSummary/TagsSection.js:167]: this is to fix a warning in the console

**Comment by mddilley** [comment]
fyi y'all - i'm reading through the [change log](https://github.com/apollographql/apollo-client/blob/main/CHANGELOG.md) to see if anything major sticks out to me that we might not catch in the release testing.

**Review by roseeichelmann — APPROVED** [review]
Yay works great, thanks for fixing this! glad they already had patched the bug and we just needed to upgrade

**Review by frankhereford — APPROVED** [review]
Works so great! Thank you for updating this library Chia and for figuring out what was going on here. Code looks awesome too -- thanks for picking off that console error as well!

🚢🚢🚢

File: moped-editor/package-lock.json [hunk 2]
```diff
@@ -115,25 +115,27 @@
       }
     },
     "node_modules/@apollo/client": {
-      "version": "3.8.9",
-      "resolved": "https://registry.npmjs.org/@apollo/client/-/client-3.8.9.tgz",
-      "integrity": "sha512-IcQDFEEPc9+PEQsxhxQvsoQ04BRarOzi/Ila5PcniRSDeKJWgY22dnp6+V1i1fWXRDVd1ybdvze4sFESDVQUCQ==",
+      "version": "3.10.4",
```
Comment (mddilley) [inline moped-editor/package-lock.json:118]: I read the changelog up to 3.10.4 which the update in the package.json resolved to here. Nothing stuck out to me! 🚀

**Review by mddilley — APPROVED** [review]
I tested moving many components locally, and I can't get this to break anymore. I also tested some of the main views, and our fetching/caching/mutating appears to be working just as before. Thank you for getting to the bottom of this, and we get one of our libraries updated along the way! 🙌 🚀

**Pull request merged**